extensions/amp-skimlinks/0.1/link-rewriter/link-rewriter.js

```diff
@@ -236,12 +253,24 @@ export class LinkRewriter {
 
   /**
    * Get the list of anchors element in the page.
-   * (Based on linkSelector option)
+   * (Based on linkSelector & excludeSelector options)
    * @return {!Array<!HTMLElement>}
    * @private
    */
   getLinksInDOM_() {
-    const q = this.rootNode_.querySelectorAll(this.linkSelector_);
-    return [].slice.call(q);
+    const includedNodeList = this.rootNode_.querySelectorAll(
+      this.linkSelector_
+    );
+    // Convert from NodeList to array
+    let anchors = nodeListToArray(includedNodeList);
+
+    if (this.excludeSelector_) {
+      const excludedAnchors = nodeListToArray(
+        this.rootNode_.querySelectorAll(this.excludeSelector_)
+      );
+      anchors = anchors.filter(a => excludedAnchors.indexOf(a) === -1);
```

Comment:
not sure how big of a deal it is, but you are not forcing the `exludeSelector` / `linkSelector` to match an actual anchor.

Reply:
It is a bit of a concern, I put a WARNING in the documentation https://amp.dev/documentation/components/amp-skimlinks/ to tell the publisher to use this option carefully. I can't think of a reliable way to validate that a CSS selector is matching an `a` element. Maybe some regex but not sure if we want to go down that road. 

For now we need to rely on the publisher being careful.

Reply:
when you are filtering anchors you can check for `.tagname` but it will add some processing time (+ you will have to do the filter even if there are no excludeSelectors